src/commands/force/source/beta/status.ts

```diff
@@ -0,0 +1,169 @@
+      command: replaceRenamedCommands('force:source:status'),
+    });
+    const wantsLocal =
+      (this.flags.local as boolean) || (this.flags.all as boolean) || (!this.flags.remote && !this.flags.all);
+    const wantsRemote =
+      (this.flags.remote as boolean) || (this.flags.all as boolean) || (!this.flags.local && !this.flags.all);
+
+    this.logger.debug(
+      `project is ${this.project.getPath()} and pkgDirs are ${this.project
+        .getPackageDirectories()
+        .map((dir) => dir.path)
+        .join(',')}`
+    );
+    const tracking = await SourceTracking.create({
+      org: this.org,
+      project: this.project,
+      apiVersion: this.flags.apiversion as string,
+    });
+
+    if (wantsLocal) {
+      await tracking.ensureLocalTracking();
+      const localDeletes = tracking.populateTypesAndNames({
+        elements: await tracking.getChanges<ChangeResult>({ origin: 'local', state: 'delete', format: 'ChangeResult' }),
+        excludeUnresolvable: true,
+        resolveDeleted: true,
+      });
+
+      const localAdds = tracking.populateTypesAndNames({
+        elements: await tracking.getChanges<ChangeResult>({ origin: 'local', state: 'add', format: 'ChangeResult' }),
+        excludeUnresolvable: true,
+      });
+
+      const localModifies = tracking.populateTypesAndNames({
+        elements: await tracking.getChanges<ChangeResult>({ origin: 'local', state: 'modify', format: 'ChangeResult' }),
+        excludeUnresolvable: true,
+      });
+
+      this.results = this.results.concat(
+        localAdds.flatMap((item) => this.statusResultToOutputRows(item, 'add')),
+        localModifies.flatMap((item) => this.statusResultToOutputRows(item, 'changed')),
+        localDeletes.flatMap((item) => this.statusResultToOutputRows(item, 'delete'))
+      );
+    }
+
+    if (wantsRemote) {
+      // by initializeWithQuery true, one query runs so that parallel getChanges aren't doing parallel queries
+      await tracking.ensureRemoteTracking(true);
+      const [remoteDeletes, remoteModifies] = await Promise.all([
+        tracking.getChanges<ChangeResult>({ origin: 'remote', state: 'delete', format: 'ChangeResult' }),
+        tracking.getChanges<ChangeResult>({ origin: 'remote', state: 'nondelete', format: 'ChangeResult' }),
+      ]);
+      this.results = this.results.concat(
+        remoteDeletes.flatMap((item) => this.statusResultToOutputRows(item)),
+        remoteModifies.flatMap((item) => this.statusResultToOutputRows(item))
+      );
+    }
+
+    if (wantsLocal && wantsRemote) {
+      // keys like ApexClass__MyClass.cls
+      const conflictKeys = (await tracking.getConflicts()).map((conflict) => getKeyFromObject(conflict));
+      if (conflictKeys.length > 0) {
+        this.results = this.results.map((row) =>
+          conflictKeys.includes(getKeyFromStrings(row.type, row.fullName))
+            ? { ...row, state: `${row.state} (Conflict)` }
+            : row
+        );
+      }
+    }
```

Comment:
it seems like lines 76-124 could be wrapped into a STL function.  Is that possible?

Reply:
done in https://github.com/salesforcecli/plugin-source/pull/258